### DISCUSSION TITLE
isConnected from useAccount() returns false after I connect my wallet

### DISCUSSION THREAD

**Discussion by krisibateva** [wevm/wagmi#1361] [Q&A] [answered]
Hello, I have just started working with wagmi and I don't understand why isConnected returns false, after I connect my wallet.

I can confirm in Metamask that my wallet is connected.

I need to include this logic in my dapp, so would appreciate any advice, this is my code:

```tsx
export const ConnectWallet = () => {
  const { isConnected } = useAccount()
  const { connect, connectors, error, pendingConnector } = useConnect()

  console.log("isConnected: ", isConnected) //returns false
  
  return (
    <>
      <button className='wallet-btn' type='button' onClick={() => connect({ connector: connectors })}>
         Connect to Polygon
      </button>
      {error && <div>{error.message}</div>}
    </>
  );
};
```

**Comment by tmm** [comment, marked as answer]
Should work fine! Can you create a link to a runnable example that reproduces this?

**Reply by krisibateva** [reply]
https://converter-dapp-ntyp3r0lp-krisibateva.vercel.app/

if this doesn't work, I can invite you to the repository

**Reply by tmm** [reply]
Minimal reproducible example, using something like CodeSandbox, much easier to take a look at.

**Reply by krisibateva** [reply]
https://codesandbox.io/s/hardcore-keller-lfvlpx?file=/src/ConnectWallet.jsx

**Reply by krisibateva** [reply]
Hi again, actually I get an error when I try to connect as well, when I click the button the returned error is "connector is already connected" and my wallet is not connected

**Reply by tmm** [reply]
This line is incorrect. You should pass a single connector (e.g. `connectors[0]`).

```tsx
onClick={() => connect({ connector: connectors })}
```

**Reply by krisibateva** [reply]
that resolved the issue, thank you
